data/basics/userBalance.go

```diff
@@ -381,6 +381,14 @@ func (u AccountData) Money(proto config.ConsensusParams, rewardsLevel uint64) (m
 	return e.MicroAlgos, e.RewardedMicroAlgos
 }
 
+// PendingRewards computes the amount of rewards (in microalgos) that
+// have yet to be added to the account balance.
+func PendingRewards(ot *OverflowTracker, proto config.ConsensusParams, microAlgos MicroAlgos, rewardsBase uint64, rewardsLevel uint64) MicroAlgos {
+	rewardsUnits := microAlgos.RewardUnits(proto)
+	rewardsDelta := ot.Sub(rewardsLevel, rewardsBase)
+	return MicroAlgos{Raw: ot.Mul(rewardsUnits, rewardsDelta)}
+}
```

Comment:
Indeed it's basically the same code.  The reason I didn't merge them is that I liked the detailed panic message that describes the internal computation steps that overflowed (in the off chance that we do trigger overflow), and the function below does one more overflow-checked addition that could overflow and cause that message to be printed (so it's not just a matter of moving the panic-on-error into this new `PendingRewards()`)..

This code is rather deeply embedded in our protocol now, so it seemed not so bad to have a copy of this code, since it's not so likely to change.